precompile/fee_config_manager.go

```diff
@@ -0,0 +1,268 @@
+
+	numFeeConfigField = (maxFeeConfigFieldKey - minFeeConfigFieldKey + 1)
+)
+
+var (
+	_ StatefulPrecompileConfig = &FeeConfigManagerConfig{}
+
+	// Singleton StatefulPrecompiledContract for setting fee configs by permissioned callers.
+	FeeConfigManagerPrecompile StatefulPrecompiledContract = createFeeConfigManagerPrecompile(FeeConfigManagerAddress)
+
+	setFeeConfigSignature = CalculateFunctionSelector("setFeeConfig(uint256,uint256,uint256,uint256,uint256,uint256,uint256,uint256)")
+	getFeeConfigSignature = CalculateFunctionSelector("getFeeConfig()")
+
+	ErrCannotChangeFee = errors.New("non-enabled cannot change fee config")
+
+	// 8 fields in FeeConfig struct
+	feeConfigInputLen = common.HashLength * numFeeConfigField
+)
+
+// FeeConfigManagerConfig wraps [AllowListConfig] and uses it to implement the StatefulPrecompileConfig
+// interface while adding in the contract deployer specific precompile address.
+type FeeConfigManagerConfig struct {
+	AllowListConfig
+}
+
+// Address returns the address of the fee config manager contract.
+func (c *FeeConfigManagerConfig) Address() common.Address {
+	return FeeConfigManagerAddress
+}
+
+// Configure configures [state] with the desired admins based on [c].
+func (c *FeeConfigManagerConfig) Configure(state StateDB) {
+	c.AllowListConfig.Configure(state, FeeConfigManagerAddress)
+}
+
+// Contract returns the singleton stateful precompiled contract to be used for the fee manager.
+func (c *FeeConfigManagerConfig) Contract() StatefulPrecompiledContract {
+	return FeeConfigManagerPrecompile
+}
+
+// GetFeeConfigManagerStatus returns the role of [address] for the fee config manager list.
+func GetFeeConfigManagerStatus(stateDB StateDB, address common.Address) AllowListRole {
+	return getAllowListStatus(stateDB, FeeConfigManagerAddress, address)
+}
+
+// SetFeeConfigManagerStatus sets the permissions of [address] to [role] for the
+// fee config manager list. assumes [role] has already been verified as valid.
+func SetFeeConfigManagerStatus(stateDB StateDB, address common.Address, role AllowListRole) {
+	setAllowListRole(stateDB, FeeConfigManagerAddress, address, role)
+}
+
+// PackGetFeeConfigInput packs the getFeeConfig signature
+func PackGetFeeConfigInput() []byte {
+	return getFeeConfigSignature
+}
+
+// PackFeeConfig packs [feeConfig] without the selector into the appropriate arguments for fee config operations.
+func PackFeeConfig(feeConfig commontype.FeeConfig) ([]byte, error) {
+	//  input(feeConfig)
+	return packHelper(feeConfig, false)
+}
+
+// PackSetFeeConfig packs [feeConfig] with the selector into the appropriate arguments for setting fee config operations.
+func PackSetFeeConfig(feeConfig commontype.FeeConfig) ([]byte, error) {
+	// function selector (4 bytes) + input(feeConfig)
+	return packHelper(feeConfig, true)
+}
+
+func packHelper(feeConfig commontype.FeeConfig, useSelector bool) ([]byte, error) {
+	fullLen := feeConfigInputLen
+	packed := [][]byte{
```

Comment:
nit: may want to create utility to take an array of big.Int and perform this transformation.

Reply:
you mean for the FeeConfig type?

Reply:
No I mean right now we are passing in `*big.Int` and sometimes `common.Hash` etc, and pack them individually, so if we had a helper that handled the following interface:

```
type Bytable interface {
     Bytes() []byte
}
```

then we could pass either in.